src/main/webapp/app/entities/flair-bi/fullscreen/flair-bi-fullscreen.controller.js

```diff
@@ -139,6 +157,113 @@
             filterParametersService.save(filterParameters);
         }
 
+
+        function applyDefaultFilters(excluded) {
+            const filterParameters = filterParametersService.get();
+            const mandatoryDimensions = featureEntities
+                .filter(function (item) {
+                    return !excluded.includes(item);
+                })
+                .filter(function (item) {
+                    return item.featureType === "DIMENSION" && item.dateFilter === "ENABLED";
+                })
+                .filter(function (item) {
+                    return !filterParameters[item.name];
+                });
+
+            if (mandatoryDimensions.length > 0) {
+                const filterParameters = filterParametersService.getSelectedFilter();
+                mandatoryDimensions.forEach(function (item) {
+                    var startDate = new Date();
+                    startDate.setDate(startDate.getDate() - 1);
+                    item.selected = filterParametersService.dateToString(startDate);
+                    item.selected2 = filterParametersService.dateToString(new Date());
+                    item.metadata = {
+                        dateRangeTab: 1,
+                        customDynamicDateRange: 1,
+                        currentDynamicDateRangeConfig: {}
+                    };
+                    delete filterParameters[item.name]
+                    addDefaultDateRangeFilter(filterParameters, item.selected, item);
+                    addDefaultDateRangeFilter(filterParameters, item.selected2, item);
+                });
+                filterParametersService.saveSelectedFilter(filterParameters);
+                filterParametersService.save(filterParametersService.getSelectedFilter());
+            }
+        }
+
+        function applyViewFeatureCriteria() {
```

Comment:
This is duplicate function exist in header..check if u can create a common service in dispatcher js file